first-analysis-steps/ganga-data.md

```diff
@@ -33,25 +33,42 @@ Ganga Out [5]: ['root://bw32-4.grid.sara.nl:1094/pnfs/grid.sara.nl/data/lhcb/LHC
 {% endcallout %} 
 
-To avoid filling up the filespace it is wise to put the large files produced by your job somewhere with lots of storage - the grid. You can do so by setting the `outputfiles` attribute:
+To avoid filling up the filespace, it is wise to put the large files produced by your job somewhere with lots of storage - the grid. You can do so by setting the `outputfiles` attribute:
 
 ```python
 j.outputfiles = [DiracFile('*.root'), LocalFile('stdout')]
 ```
-The `DiracFile` will be stored in your user area on the grid (with up to 2TB personal capacity) from where you can access it with the `accessURL()` function as before. The wildcard means that any root file produced by your job will stay on the grid. `LocalFile` downloads the file to your `gangadir`, in this case the one called `stdout`.
+The `DiracFile` will be stored in your user area on the grid (with up to 2TB personal capacity). The wildcard means that any root file produced by your job will stay on the grid. `LocalFile` downloads the file to your `gangadir`, in this case the one called `stdout`.
+You can access your files stored on the grid with the `accessURL()` function as before. 
+For example, to access the location of the output `.root` file of a specific subjob, one can use 
 
-Small files are downloaded as standard: `.root`, logfiles etc. Files that are expected to be large which have extensions .dst etc are by default kept on the grid as Dirac files. In general you are encouraged to keep your large files on the grid to avoid moving large amounts of data around through your work area.
+```python
+jobs(787).subjobs(15).outputfiles[0].accessURL()
+```
+This is a very useful feature: you do not have to download your files from the grid in order to merge them locally! This takes a lot of time and disk space.
+Instead, one can get a list of URLs from each subjob, and pass them to the `TChain` or use the `hadd` method of ROOT[^1]. 
+Ganga has a shortcut to access the list of all `.root` files from all the subjobs of a given job:
+
+```python
+jobs(787).backend.getOutputDataAccessURLs()
+```
+
+Small files are downloaded as standard: `.root`, logfiles etc. Files that are expected to be large (with extensions `.dst` etc) are by default kept on the grid as Dirac files. In general, you are encouraged to keep your large files on the grid to avoid moving large amounts of data around through your work area. 
+
+More information on the `DiracFile` usage is provided in the [next lesson](eos-storage.html).
 
 
 {% callout "Getting help with ganga" %}
 To find out more take a look at the [Ganga 
 FAQ](https://twiki.cern.ch/twiki/bin/view/LHCb/FAQ/GangaLHCbFAQ)
 {% endcallout %} 
+
+[^1]: By default, DaVinci uses the compression level 6 for the `.root` files, which is different from the default compression level in ROOT. Merging your files with `hadd` will be significantly faster if you run it with the option telling to use the compression level 6 also for the output file: `hadd -f6`.
```

Comment:
As this might change or people can get ROOT files from other sources I think this should suggest using the generic `-fk` option.

```suggestion
[^1]: Merging your files with `hadd` will be significantly faster if you run it with the option telling ROOT to use the same compression level in that output file as is used for the input files. This can be done using the `-fk` option. If running on lxplus you will need to get a newer ROOT version that supports this option by using: `lb-run ROOT hadd -fk output.root input/*.root`
```

Reply:
AFAIK `-fk` option is not available in ROOT5. Probably we can mention both?

Reply:
It's not available in ROOT 5 (or some older versions of ROOT 6). Is there any reason for people to use ROOT 5 other than it's the default on lxplus?

I realise now that the suggest change feature make it's it hard to read wrong lines, but I mention using `lb-run ROOT hadd` on lxplus to avoid this issue.

 > [^1]: Merging your files with `hadd` will be significantly faster if you run it with the option telling ROOT to use the same compression level in that output file as is used for the input files. This can be done using the `-fk` option. If running on lxplus you will need to get a newer ROOT version that supports this option by using: `lb-run ROOT hadd -fk output.root input/*.root`

Reply:
I would naively expect people trying this in the default lxplus shell. But your solution with `lb-run ROOT` looks promising, so let's stick to this!